Lib/urllib/robotparser.py

```diff
@@ -56,10 +56,10 @@ def set_url(self, url):
         self.url = url
         self.host, self.path = urllib.parse.urlparse(url)[1:3]
 
-    def read(self):
+    def read(self, sslcontext=None):
         """Reads the robots.txt URL and feeds it to the parser."""
         try:
-            f = urllib.request.urlopen(self.url)
+            f = urllib.request.urlopen(self.url, context=sslcontext) if sslcontext else urllib.request.urlopen(self.url)
```

Comment:
context is already set to None by default in urlopen() so no need to check if sslcontext is None or not.